client/jetpack-connect/controller.js

```diff
@@ -107,6 +108,23 @@ export function saveQueryObject( context, next ) {
 	next();
 }
 
+export function maybeOnboard( context, next ) {
+	if ( ! isEmpty( context.query ) && context.query.onboarding ) {
+		const siteId = context.query.client_id;
+		const credentials = {
+			token: context.query.onboarding,
+			siteUrl: context.query.site_url,
+			userEmail: context.query.user_email,
+		};
+
+		context.store.dispatch( receiveJetpackOnboardingCredentials( siteId, credentials ) );
+
+		return page.redirect( '/jetpack/onboarding' );
```

Comment:
Hmm, in the long run, we'll want to redirect to `'/jetpack/onboarding/' + siteSlug`. Do we have `siteSlug` available somewhere in `context`? Or should we simply use `siteId`?

Reply:
> Hmm, in the long run, we'll want to redirect to `'/jetpack/onboarding/' + siteSlug`. 

Good point. Let's decide whether we'll need to onboad multiple sites and handle that in a separate PR I guess, because we should also adapt it for the steps too.

> Do we have `siteSlug` available somewhere in `context`? 

Nope, we have `site_url` and we can extract it from there.

> Or should we simply use `siteId`?

I have no preferences towards using `siteId` or `siteSlug` in the routes.